pkg/acceptor/quic_acceptor.go

```diff
@@ -0,0 +1,244 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package acceptor
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/quic-go/quic-go"
+	"github.com/topfreegames/pitaya/v3/pkg/conn/codec"
+	"github.com/topfreegames/pitaya/v3/pkg/conn/packet"
+	"github.com/topfreegames/pitaya/v3/pkg/constants"
+)
+
+var (
+	// ErrListenerNotInitialized is returned if the QUIC listener is not initialized
+	ErrListenerNotInitialized = errors.New("listener not initialized")
+	// ErrConnClosed is returned when the QUIC connection is closed
+	ErrConnClosed = errors.New("connection is closed")
+)
+
+// QuicAcceptor represents a QUIC acceptor
+type QuicAcceptor struct {
+	addr     string
+	connChan chan PlayerConn
+	listener *quic.Listener
+	running  bool
+	tlsConf  *tls.Config
+	quicConf *quic.Config
+}
+
+// NewQuicAcceptor creates a new QuicAcceptor
+func NewQuicAcceptor(addr string, tlsConf *tls.Config, quicConf *quic.Config) *QuicAcceptor {
+	return &QuicAcceptor{
+		addr:     addr,
+		tlsConf:  tlsConf,
+		quicConf: quicConf,
+		connChan: make(chan PlayerConn),
+	}
+}
+
+func (a *QuicAcceptor) GetAddr() string {
+	if a.listener != nil {
+		return a.listener.Addr().String()
+	}
+	return ""
+}
+
+// GetConnChan gets a connection channel
+func (a *QuicAcceptor) GetConnChan() chan PlayerConn {
+	return a.connChan
+}
+
+// Listen starts the QUIC listener to accept new connections
+func (a *QuicAcceptor) Listen() error {
+	// Creating a QUIC listener at the specified address
+	a.running = true
+	listener, err := quic.ListenAddr(a.addr, a.tlsConf, a.quicConf)
+	if err != nil {
+		return err
+	}
+	a.listener = listener
+	return nil
+}
+
+// Accept accepts new QUIC connections
+func (a *QuicAcceptor) Accept() (quic.Connection, error) {
+	if a.listener == nil { // Correct comparison
+		return nil, ErrListenerNotInitialized
+	}
+
+	conn, err := a.listener.Accept(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+// Close closes the QUIC listener
+func (a *QuicAcceptor) Close() error {
+	if a.listener != nil { // Correct comparison
+		return a.listener.Close()
+	}
+	return nil
+}
+
+// EnableProxyProtocol not implemented for QUIC, keep as No-op or implement as needed
+func (a *QuicAcceptor) EnableProxyProtocol() {
```

Comment:
@guthyerrz I think it's important to understand who's using proxy protocol and what's the use case, so we evaluate if we need to implement this right away or not